### ISSUE TITLE
Misleading warning when using dynamic property names to use API

### ISSUE THREAD

**Issue by freaktechnik** [mozilla/addons-linter#1309]
### Describe the problem and steps to reproduce it:
Use an API with a dynamic property accessor, like the following:
```js
const AREA = "local";
browser.storage[AREA].set({
  migrated: true
});
```

### What happened?
The warning printed by the linter is "storage.AREA is not supported".

### What did you expect to happen?
A warning along the lines of "Could not determine if API is implemented for storage[AREA]"

### Anything else we should know?
Source code with which I have experienced this: https://github.com/freaktechnik/justintv-stream-notifications/blob/master/src/background/preferences.js
Travis CI build with the warnings: https://travis-ci.org/freaktechnik/justintv-stream-notifications#L1971-L1976

**Comment by muffinresearch** [comment]
@mstriemer any thoughts on how the schema can be used to not trip over this sort of thing?

**Comment by mstriemer** [comment]
I don't think we can check that a usage like this is valid. We could warn that we can't validate, but I just put a PR up to ignore the check in this case (#1332). Warning might be better.

@muffinresearch should we warn or ignore here?

**Comment by vcarciu** [comment]
Please add some STRs to this bug or mark it as "qa: not needed"
